simple_database/dbmodule.py

```diff
@@ -0,0 +1,124 @@
+import json, os, pickle
+try:
+    from exceptions import *
+    from config import BASE_DB_FILE_PATH
+except:
+    from .exceptions import *
+    from .config import BASE_DB_FILE_PATH
+
+
+def initDbIndex(): #This function takes care of creating the index file, if it doesn't exist
+    if not os.path.exists(BASE_DB_FILE_PATH):
+        os.makedirs(BASE_DB_FILE_PATH)
+    if not os.path.isfile(os.path.join(BASE_DB_FILE_PATH, "path_index.json")):
+        open(os.path.join(BASE_DB_FILE_PATH, "path_index.json"), "a").close()
+        
+def create_database(db_name):
+    initDbIndex() #Check if the index file exists, or take care of it
+    if Db.doesDbExist(db_name):
+        raise ValidationError('Database with name "{}" already exists.'.format(db_name))
+    else:#If the db doesn't exist
+        db = Db(db_name)#Create it
+        with open(os.path.join(BASE_DB_FILE_PATH, "path_index.json"), "r") as file: #And update the index file
+            temp_file = file.read()
+            if temp_file: temp_file = json.loads(temp_file)
+            else: temp_file = {}
+            temp_file[db_name] = db.path
+        with open(os.path.join(BASE_DB_FILE_PATH, "path_index.json"), "w") as file:
+            file.write(json.dumps(temp_file))
+        return db
+        
+    # raise NotImplementedError()
+
+def connect_database(db_name):
+    db_connect = Db.doesDbExist(db_name)
+    if not db_connect:
+        raise ValidationError("Database does not exist")
+    else: #If the db exists
+        with open(os.path.join(BASE_DB_FILE_PATH, "path_index.json"), "r") as index_file: #Obtain its path from the index file
+            index = json.loads(index_file.read())
+            with open(os.path.join(BASE_DB_FILE_PATH, index[db_name]), "rb") as pickled_file:#And load it from hard drive
+                db = pickle.load(pickled_file)
+                return db
+ 
+        
+    #error if doesnt exist
+
+class Table(object):
+    def __init__(self, db, name, columns):
+        self.db = db
+        self.name = name
+        self.columns = columns
+        self.rows = []
+        self.start = 0
+    
+    
+    def describe(self):
+        return self.columns
+
+    def count(self):
+        return len(self.rows)   
+
+    def all(self):
+        return ( elem for elem in self.rows )
+        
+    def query(self, **kwarg):  #key=value
+        results = []
+        for row in self.rows:
+            if list(kwarg.values())[0] in getattr(row, list(kwarg.keys())[0]):
+                results.append(row)
+        return iter(results)        
+    
+    def insert(self, *args): 
+        if len(args) != len(self.columns):
+            raise ValidationError("Invalid amount of field")
+        for index, arg in enumerate(args):
+            if type(arg).__name__ != self.columns[index]['type']:
```

Comment:
Wow, that's great. After we discussed how everything in Python is an object, too. We never thought to look for the name of the 'int' object.